runtime/appruntime/reqtrack/reqtrack.go

```diff
@@ -42,11 +42,32 @@ func (t *RequestTracker) FinishOperation() {
 
 func (t *RequestTracker) BeginRequest(req *model.Request) {
 	if prev, _, _ := t.currentReq(); prev != nil {
+		copyReqInfoFromParent(req, prev)
 		t.clearReq()
 	}
 	t.beginReq(req, req.Traced)
 }
 
+// copyReqInfoFromParent copies over relevant request from the parent request.
+// If the relevant fields are already set on next they are not copied over.
+func copyReqInfoFromParent(next, prev *model.Request) {
+	if next.UID == "" {
+		next.UID = prev.UID
+	}
+	if next.AuthData == nil {
+		next.AuthData = prev.AuthData
+	}
+	if next.ParentID == (model.SpanID{}) {
+		next.ParentID = prev.SpanID
+	}
+	if !next.Traced {
+		next.Traced = prev.Traced
+	}
+	if next.Test == nil {
+		next.Test = prev.Test
+	}
```

Comment:
Should we copy over the pubsub object? So you can introspec to see if this is an interactive client calling or a background client calling?

Reply:
I think if a sub-request happens we shouldn't surface the pubsub data in child requests, or what do you mean?